src/main.rs

```diff
@@ -214,44 +320,140 @@ fn move_player(keyboard_input: Res<Input<KeyCode>>, mut query: Query<&mut Veloci
+                    },
+                    ..Default::default()
+                })
+                .insert(RigidBody::Sensor)
+                .insert(Velocity::from_linear(Vec3::default()))
+                .insert(BasicAttack)
+                .insert(Delay {delay: 0.1, start: Instant::now()})
+                .insert(CollisionLayers::new(Layer::Projectile, Layer::Enemy));
         }
+    }
+}
 
-        if keyboard_input.pressed(KeyCode::W) {
-            direction += Vec3::new(0., -SPEED, 0.);
+fn end_attack(mut commands: Commands, query: Query<(&BasicAttack, Entity, &Delay)>) {
+    for (_, entity, delay) in query.iter() {
+        if delay.next_action_aviable(Instant::now()) {
+            commands.entity(entity).despawn();
         }
+    }
+}
+
+
+fn move_slime(mut query: Query<(&mut Velocity, &mut GlobalTransform, &Speed), (With<Slime>, Without<Player>)>, 
+    player_query: Query<&GlobalTransform, (With<Player>, Without<Slime>)>) {
+    for (mut velocity, slime_pos, speed) in query.iter_mut() {
+        for player_pos in player_query.iter() {
+            let mut direction = Vec3::new(0., 0., 0.);
 
-        if keyboard_input.pressed(KeyCode::S) {
-            direction += Vec3::new(0., SPEED, 0.);
+            if slime_pos.translation.x > player_pos.translation.x {
+                direction += Vec3::new(-1.0, 0., 0.);
+            }
+    
+            if slime_pos.translation.x < player_pos.translation.x {
+                direction += Vec3::new(1.0, 0., 0.);
+            }
+    
+            if slime_pos.translation.y > player_pos.translation.y {
+                direction += Vec3::new(0., -1.0, 0.);
+            }
+    
+            if slime_pos.translation.y < player_pos.translation.y {
+                direction += Vec3::new(0., 1.0, 0.);
+            }
+
+            direction = direction.normalize_or_zero();
+
+            *velocity = Velocity::from_linear(direction * speed.value);
+        }
+
+    }
+}
+
+fn move_flame_spirit(mut query: Query<(&mut Velocity, &mut GlobalTransform, &Speed, &mut Delay), (With<FlameSpirit>, Without<Player>)>, 
+    player_query: Query<&GlobalTransform, (With<Player>, Without<FlameSpirit>)>) {
+    for (mut velocity, flame_pos, speed, mut delay) in query.iter_mut() {
+        for player_pos in player_query.iter() {
+            let mut direction = Vec3::new(0., 0., 0.);
+
+            if flame_pos.translation.x > player_pos.translation.x {
+                direction += Vec3::new(-1.0, 0., 0.);
+            }
+    
+            if flame_pos.translation.x < player_pos.translation.x {
+                direction += Vec3::new(1.0, 0., 0.);
+            }
+    
+            if flame_pos.translation.y > player_pos.translation.y {
+                direction += Vec3::new(0., -1.0, 0.);
+            }
+    
+            if flame_pos.translation.y < player_pos.translation.y {
+                direction += Vec3::new(0., 1.0, 0.);
+            }
+
+            direction = direction.normalize_or_zero();
```

Comment:
This thing is copied a lot with small variation, probably it is a good idea to find a way to make a function which can make the changes and normalize at the end and use that